_source/_posts/2021-05-28-using-okta-and-oso-to-secure-a-fastapi-and-sqlalchemy-app.md

```diff
@@ -0,0 +1,496 @@
+---
+layout: blog_post
+title: "Using Okta + Oso to secure a FastAPI + SQLAlchemy app"
+author: gabe-jackson
+by: contractor
+communities: [python]
+description: "How to use Oso with Okta to add authorization to a FastAPI and SQLAlchemy app"
+tags: [oso, authorization]
+tweets:
+- "Learn how to add simple authorization to your app with Oso and Okta"
+- "Check out this blog post to learn how to easily secure your FastAPI app with Oso and Okta"
+- "Easily add role based access to your FastAPI app with Oso and Okta"
+image: blog/using-okta-and-oso-to-secure-a-fastapi-and-sqlalchemy-app/card.png
+type: conversion
+---
+FastAPI is really fast and SQLAlchemy is really... SQL-y, but what good is a fast and SQL-y application if it isn't 🔒**secure**🔒?
+
+In this post, we're going to show you how to secure a fast and SQL-y app!
+
+First we will need some _authentication_, which is how we identify
+**who** the user is. We'll use Okta for this.
+
+Next, we'll want to perform _authorization_, which controls **what** the user can do in our application. We'll be using [Oso][] for that, which is a batteries-included library for authorization.
+
+This post is aimed at people with some familiarity with FastAPI and SQLAlchemy. By the end, you'll be able to make sure users have access to the things they need - and only the things they need.
+
+The full example is available on [GitHub][]. Clone the repo and follow along!
+
```

Comment:
Add a table of contents so you can see if the headers have good SEO for keywords.

```
**Table of Contents**{: .hide }
* Table of Contents
{:toc}
```